webextensions/sidebar/sidebar-tabs.js

```diff
@@ -41,6 +43,13 @@ export const wholeContainer = document.querySelector('#all-tabs');
 export const onSyncFailed = new EventListenerManager();
 
 export function init() {
+  const elements = [
+    [Constants.kTAB_CLOSE_BOX_ELEMENT, TSTCloseBoxElement],
+  ];
+  for (const [localName, constructor] of elements) {
+    window.customElements.define(localName, constructor);
+  }
+
```

Comment:
Why you didn't put these codes in the `TSTCloseBoxElement.js`? It looks better for me to define the relation of `Constants.kTAB_CLOSE_BOX_ELEMENT` and `TSTCloseBoxElement` in the `TSTCloseBoxElement.js` itself. If there is any reason why we shouldn't do that, I need to understand it well. My understanding is:

* The element name may be changed in cases, so we should not tie them as a pair in general.
* When I reuse these codes partially for other addons, tightly paired element name may block us to do that.

If they are full reasons of this design, then I still think that the element constructor class can have its default element name as something like its public property or others.

Reply:
> Why you didn't put these codes in the TSTCloseBoxElement.js?

If we call `window.customElements.define(localName, constructor);` from the file defining a custom element, it would be a side effect and happen accidentally that defining a custom element when we import a new file which defines a new custom element. It causes a complex side effect relations. 

I have not concluded the best practice about it yet, but I think that it's safely to call window.customElements.define(localName, constructor);` separately in the application initialization phase.

> The element name may be changed in cases, so we should not tie them as a pair in general.

I did not intend to make a local name and a class loose coupled.

I'll add the change to move `Constants.kTAB_CLOSE_BOX_ELEMENT` to the file which defines the custom element.

Reply:
OK, I agree that accidental side effect is not good and `window.customElements.define()` should be here. 

> I'll add the change to move Constants.kTAB_CLOSE_BOX_ELEMENT to the file which defines the custom element.

Thanks. Defining the constant for the element name in the same file looks reasonable. I'm imagine that it is accessible like `TabCloseBoxElement.ELEMENT_NAME` or something, but it is just a matter of personal preference. (But I can't understand why the spec doesn't provide a way to resolve the default element name from the class name itself automatically...)

Reply:
@piroor 

I updated the patch https://github.com/piroor/treestyletab/pull/2392/commits/21f4fa9c1fa09784cf8c1b97ee76f2eb846fbdb1

> I think that it's rare case that we need static properties/methods for class in today's ECMAScript.

I believe we fulfill this "rare case" by both of them

1. We can provide an ergonomic API.
2. A provided operation/values has a strong relation to its data class.